src/send.c

```diff
@@ -35,33 +35,31 @@ static inline void add_pktinfo(struct msghdr *msg, const fastd_peer_address_t *l
 
 #ifdef USE_PKTINFO
 	if (local_addr->sa.sa_family == AF_INET) {
+		struct in_pktinfo *pktinfo = (struct in_pktinfo *)CMSG_DATA(cmsg);
```

Comment:
In many libc's (for example musl) `cmsghr` doesn't have a flexible member for the data and `CMSG_DATA()` is defined as follows:
```
#define CMSG_DATA(cmsg) ((unsigned char *) (((struct cmsghdr *)(cmsg)) + 1))
```
Casting this to `struct in_pktinfo *` is an aliasing violation and memcpy should be used.